src/backend/aspen/app/app.py

```diff
@@ -70,13 +77,59 @@
 )
 
 
+def validate_auth_header(auth_header):
+    parts = auth_header.split()
+
+    if parts[0].lower() != "bearer":
+        raise TokenValidationError("Authorization header must start with Bearer")
+    elif len(parts) == 1:
+        raise TokenValidationError("Token not found")
+    elif len(parts) > 2:
+        raise TokenValidationError("Authorization header must be Bearer token")
+
+    id_token = parts[1]
+
+    domain = application.aspen_config.AUTH0_DOMAIN
+    client_id = application.aspen_config.AUTH0_CLIENT_ID
+
+    jwks_url = f"https://{domain}/.well-known/jwks.json"
+    issuer = f"https://{domain}/"
+
+    # Adapted from https://github.com/auth0/auth0-python#id-token-validation
+    sv = AsymmetricSignatureVerifier(jwks_url)  # Reusable instance
+    payload = sv.verify_signature(id_token)
+    tv = TokenVerifier(signature_verifier=sv, issuer=issuer, audience=client_id)
+    tv._verify_payload(payload)
+    return payload
+
+
+def setup_userinfo(user_id):
+    user = get_usergroup_query(g.db_session, user_id).one()
+    g.auth_user = user
+
+
 # use this to wrap protected views
 def requires_auth(f):
     @wraps(f)
     def decorated(*args, **kwargs):
-        if "profile" not in session:
-            # Redirect to Login page
+        auth_header = request.headers.get("authorization")
+        auth0_user_id = None
```

Comment:
This if/elif switches between our two authorization flows (bearer token or session profile)